### PR TITLE
Mass update to replace distinct snippets for SPF or DMARC checks

### PR DESCRIPTION
**Pull request by morriscode**

### CODE REVIEW

**Comment by morriscode** [comment]
This will need mimic override/force merge. The historical samples don't have the new values.

**Comment by morriscode** [comment]
/mql-mimic-exempt: 22804,19340,19339,19317,18463,8797,132517,136621,136706,212349,130491,222803,250166,272601,275987,240136,296917,172669,222764,249749,346807,357147,133104,414742


These are failures in our old SPF logic. They regressed on one rule, but others fire.

**Comment by morriscode** [comment]
/mql-mimic-exempt: 22804,19340,19339,19317,18463,8797,132517,136621,136706,212349,130491,222803,250166,272601,275987,240136,296917,172669,222764,249749,346807,357147,133104,414742

**Comment by morriscode** [comment]
/mql-mimic-exempt:  136730, 136731, 136732, 136729, 136733, 136736, 136734, 136735, 136740, 136738, 136739, 136737

The above are flagging when I manually run the proposed rule against them, but still showing as a regression in mimic.

**Comment by morriscode** [comment]
/mql-mimic-exempt:  181157, 185788, 167254, 167258, 323179, 315071, 262118, 358103, 413898, 416049, 416048, 416047

These are all docusign FP's that were previously already remediated, not sure why they're showing up now.

**Comment by morriscode** [comment]
- 250234 

- 250233 

both flag, both any_html rules, even though they show regressions. I'm guessing this is because they're rather large files 382,000 blank lines before a very small script block.


- 308820 
- 315818 

Both flag the rule they claim to have regressed on. Again very large files here. 

- 297200

This is an FP, newsletter, has 174 links. 

- 297272
Also an FP. 

- 331685 
also flags manually, large BMP attachment 

- 408468 
This is failing for reasons outside of this change. We're not verifying authentication on excluded sender domains.

**Comment by morriscode** [comment]
/mql-mimic-exempt: 250234, 250233, 308820, 315818, 297200, 297272, 331685, 408468

see above for reason

File: detection-rules/headers_freemail_replyto_returnpath_mismatch.yml [hunk 1]
```diff
@@ -11,21 +11,30 @@ source: |
...
   )
   and sender.email.domain.root_domain not in $free_email_providers
   
   // negate gmail autoforwards and null return paths
   and (
     headers.return_path.email is null
-    or not any([headers.return_path.email], strings.ilike(headers.return_path.local_part, "*+caf_=*"))
+    or not any([headers.return_path.email],
+               strings.ilike(headers.return_path.local_part, "*+caf_=*")
+    )
```
Comment (rw-access) [inline detection-rules/headers_freemail_replyto_returnpath_mismatch.yml:37]:
```suggestion
    or not strings.ilike(headers.return_path.local_part, "*+caf_=*")
```

File: detection-rules/impersonation_github.yml [hunk 2]
```diff
@@ -35,15 +35,11 @@ source: |
     'bithub.email'
   )
 
-  // negate highly trusted sender domains unless they fail DMARC authentication
+   // negate highly trusted sender domains unless they fail DMARC authentication
```
Comment (rw-access) [inline detection-rules/impersonation_github.yml:38]:
```suggestion
  // negate highly trusted sender domains unless they fail DMARC authentication
```

File: detection-rules/impersonation_sharepoint_fake_file_share.yml [hunk 3]
```diff
@@ -31,15 +31,11 @@ source: |
     "aka.ms"
   )
   
-  // negate highly trusted sender domains unless they fail DMARC authentication
+   // negate highly trusted sender domains unless they fail DMARC authentication
```
Comment (rw-access) [inline detection-rules/impersonation_sharepoint_fake_file_share.yml:34]:
```suggestion
  // negate highly trusted sender domains unless they fail DMARC authentication
```

File: detection-rules/impersonation_facebook.yml [hunk 4]
```diff
@@ -62,15 +62,11 @@ source: |
...
-            strings.ilike(.authentication_results.dmarc, "*fail")
-        )
-      )
+      and not headers.auth_summary.dmarc.pass
     )
     or sender.email.domain.root_domain not in $high_trust_sender_root_domains
-    
-      // salesforce has been abused for meta phishing campaigns repeatedly 
+
+    // salesforce has been abused for meta phishing campaigns repeatedly
```
Comment (rw-access) [inline detection-rules/impersonation_facebook.yml:69]:
```suggestion
   // salesforce has been abused for meta phishing campaigns repeatedly 
```

**Comment by morriscode** [comment]
/mql-mimic-exempt: 205789, 413688, 413688, 413688, 427148, 443352, 440956, 440956

**Comment by morriscode** [comment]
205789- Meta brand impersonation fires, this does not fire the previous iteration of this rule either, so not a regression caused by this PR. 

413688-Sender dom was added to high trust

427148- Not firing as sender profile is now "Uncommon".

443352-  This was a happy accident with Distinct checks

440956 - NLU Regression cred_theft, impersonation rule still fires

**Review by rw-access — APPROVED** [review]

**Pull request merged**